builtin/providers/google/resource_compute_vpn_tunnel.go

```diff
@@ -177,3 +180,89 @@ func resourceComputeVpnTunnelDelete(d *schema.ResourceData, meta interface{}) er
 
 	return nil
 }
+
+// validatePeerAddr returns false if a tunnel's peer_ip property
+// is invalid. Currently, only addresses that collide with RFC
+// 5735 (https://tools.ietf.org/html/rfc5735) fail validation.
+func validatePeerAddr(i interface{}, val string) ([]string, []error) {
+	ip := net.ParseIP(i.(string))
+	if ip == nil {
+		return nil, []error{fmt.Errorf("could not parse %q to IP address", val)}
+	}
+	for _, test := range invalidPeerAddrs {
+		if bytes.Compare(ip, test.from) >= 0 && bytes.Compare(ip, test.to) <= 0 {
+			return nil, []error{fmt.Errorf("address is invalid (is between %q and %q, conflicting with RFC5735)", test.from, test.to)}
+		}
+	}
+	return nil, nil
+}
```

Comment:
Just a style nit, but we usually use the [named result parameters](https://golang.org/doc/effective_go.html#named-results) for these validatefuncs.

```go
func validatePeerAddr(i interface{}, val string) (ws []string, es []error) {
  ip := net.ParseIP(i.(string))
  if ip == nil {
    es = append(es, fmt.Errorf("..."))
    return
  }
  for _, test := range invalidPeerAddrs {
    if ... {
      es = append(es, fmt.Errorf("..."))
      break // i think only one collision is possible yeah?
    }
  }
  return
}
```